### PR TITLE
#1475 Collapsable sidebar

### LINKED ISSUE #1475: Make Sidebar in Page Editor collapsable/minimizable
**Context**
- The sidebar is useful for switching between extensions, adding extensions, etc. 
- It's not useful while editing except for showing the "Unsaved Changes" indicator

**Wireframes**

Collapsed state: 
- In the collapsed state, the icon does not open the options page
- See our side bar for the carat icons/hover CSS

image

### PR DESCRIPTION
**Pull request by BALEHOK**
Closes #1475 

image

image

### CODE REVIEW

**Comment by fregante** [comment]
Note: the diff is huge, npm is having issues

**Comment by BALEHOK** [comment]
> Note: the diff is huge, npm is having issues

It is.
Btw, `resolutions` in `package.json` is not something that is officially supported by npm.
https://stackoverflow.com/questions/52416312/npm-equivalent-of-yarn-resolutions
https://github.com/npm/statusboard/issues/343

@twschiller I believe should not use (or try to use) unsupported and unstable features and remove it or migrate to `yarn`.

**Comment by fregante** [comment]
https://github.com/pixiebrix/pixiebrix-extension/issues/1549#issuecomment-932056179

File: package.json [hunk 1]
```diff
@@ -108,9 +108,9 @@
     "react-select-virtualized": "^3.5.0",
     "react-shadow-root": "^6.1.0",
     "react-spinners": "0.9.0",
-    "react-split-pane": "^0.1.92",
     "react-table": "^7.7.0",
     "react-toast-notifications": "^2.4.4",
+    "react-transition-group": "^4.4.2",
```
Comment (fregante) [inline package.json:113]: @BALEHOK once https://github.com/pixiebrix/pixiebrix-extension/pull/1553 is merged, you can try reverting the changes to the lockfile, using `git merge origin/main`, and then running `npm install` to update the lockfile

**Review by BLoe — CHANGES_REQUESTED** [review]
Hey, I'm seeing a UI issue where the footer doesn't pin to the bottom of the side bar:

image

**Review by BLoe — APPROVED** [review]

**Pull request merged**